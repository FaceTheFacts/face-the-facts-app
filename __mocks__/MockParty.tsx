import {ApiParty, ApiPartyStyle} from '../src/logic/api';

export const MockPartyStyle: ApiPartyStyle = {
  id: 0,
  display_name: 'test-party',
  foreground_color: '#FFFFFF',
  background_color: '#5D6265',
};

export const MockParty: ApiParty = {
  id: 0,
  label: 'test-party',
  party_style: MockPartyStyle,
};
