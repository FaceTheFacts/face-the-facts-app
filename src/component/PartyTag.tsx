import React from 'react';
import Tag from './utils/Tag';
import {StyleProp, ViewStyle} from 'react-native';
import {ApiPartyStyle} from '../logic/api';

export interface PartyTagProps {
  style?: StyleProp<ViewStyle>;
  party: ApiPartyStyle;
}

const PartyTag = ({style, party}: PartyTagProps) => {
  return (
    <Tag
      style={style}
      content={party?.display_name}
      foregroundColor={party.foreground_color}
      backgroundColor={party.background_color}
      borderColor={party.border_color}
      bold
    />
  );
};

export default PartyTag;
