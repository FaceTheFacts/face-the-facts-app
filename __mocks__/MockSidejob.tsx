import {MockParty} from './MockParty';

export const MockSidejobDashboardView = {
  politicianId: 0,
  politicianName: 'test-politician',
  party: MockParty,
  date: '2020-01-01',
  label: 'test-label',
  organization: 'test-organization',
  income: 'test-income',
  fromView: true,
};

export const MockSidejobDashboard = {
  politicianId: 0,
  politicianName: 'test-politician',
  party: MockParty,
  date: '2020-01-01',
  label: 'test-label',
  organization: 'test-organization',
  income: 'test-income',
};

export const MockSidejobProfile = {
  date: '2020-01-01',
  label: 'test-label',
  organization: 'test-organization',
  income: 'test-income',
};
