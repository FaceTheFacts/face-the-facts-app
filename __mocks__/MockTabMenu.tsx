import {TabMenuItem, TabMenuProps} from '../src/component/TabMenu';
import {
  HomeIcon,
  HomeIconSolid,
  PoliticiansIcon,
  PoliticiansIconSolid,
  ScannerIcon,
  ScannerIconSolid,
} from '../src/icons';

export const MockTabMenuItem: TabMenuItem[] = [
  {
    name: 'testname1',
    icons: [HomeIcon, HomeIconSolid],
    label: 'Test1',
  },
  {
    name: 'testname2',
    icons: [ScannerIcon, ScannerIconSolid],
    label: 'Test2',
  },
  {
    name: 'testname3',
    icons: [PoliticiansIcon, PoliticiansIconSolid],
    label: 'Test3',
  },
];

export const MockTabMenu: TabMenuProps = {
  items: MockTabMenuItem,
  selected: 'Test1',
  onSelect: jest.fn(),
  blur: false,
};
