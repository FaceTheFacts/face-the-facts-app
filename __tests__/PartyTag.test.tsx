import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import PartyTag from '../src/component/PartyTag';
import {MockParty} from '../__mocks__/MockParty';

it('Party Tag renders correctly', () => {
  const tree = renderer.create(<PartyTag party={MockParty} />).toJSON();
  expect(tree).toMatchSnapshot();
});
