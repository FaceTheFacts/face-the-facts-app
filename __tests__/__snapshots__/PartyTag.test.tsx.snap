// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Party Tag renders correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "flex-start",
      "backgroundColor": "rgba(248, 248, 248, 0.12)",
      "borderColor": undefined,
      "borderRadius": 4,
      "borderWidth": undefined,
      "padding": 5,
      "width": undefined,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#F8F8F8",
        "fontFamily": "Inter",
        "fontSize": 13,
        "fontWeight": "600",
        "marginLeft": 0,
        "marginRight": 0,
        "textTransform": "none",
      }
    }
  />
</View>
`;
